providers/src/airflow/providers/edge/executors/edge_executor.py

```diff
@@ -183,16 +183,7 @@ def _purge_jobs(self, session: Session) -> bool:
         jobs: list[EdgeJobModel] = (
             session.query(EdgeJobModel)
             .with_for_update(skip_locked=True)
-            .filter(
-                EdgeJobModel.state.in_(
-                    [
-                        TaskInstanceState.RUNNING,
-                        TaskInstanceState.SUCCESS,
-                        TaskInstanceState.FAILED,
-                        TaskInstanceState.REMOVED,
-                    ]
-                )
-            )
+            .filter(EdgeJobModel.state.isnot(TaskInstanceState.QUEUED))
```

Comment:
MySQL is failing and seems it does not like the "for update" lock in conjunction with negative filter... anyway I'd propose to stay with a positive filter like this... (as you are adding two states)
```suggestion
            .filter(
                EdgeJobModel.state.in_(
                    [
                        TaskInstanceState.RUNNING,
                        TaskInstanceState.SUCCESS,
                        TaskInstanceState.FAILED,
                        TaskInstanceState.REMOVED,
                        TaskInstanceState.RESTARTING,
                        TaskInstanceState.UP_FOR_RETRY,
                    ]
                )
            )```